### PR TITLE
PLAT-1156 Add several checks to DbTableValidator

### PR DESCRIPTION
**Pull request by alexander-schmidt**

### CODE REVIEW

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/validator/DbTableValidator.java [hunk 1]
```diff
@@ -3,71 +3,145 @@
...
-		this.dataFields = this.table.getDataFields();
+		this.dataFields = table.getDataFields();
+		this.allKeys = table.getAllKeys();
 	}
 
 	public void validate() {
 
-		validateStringFields();
-		validateNullability();
+		var errors = new ErrorList();
```
Comment (oliver-richers) [inline platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/validator/DbTableValidator.java:33]: hmm, would be less code if me made `errors` a field...just saying

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/validator/DbTableValidator.java [hunk 2]
```diff
@@ -3,71 +3,145 @@
...
+				errors
+					.add(//
 						"Field %s.`%s` defaults to NULL but it is not NULLABLE.",
 						table.getFullName(),
-						field.getTitle());
-				errors.add(message);
+						field.getName());
+			}
+
+			if (field.isNullable() && !field.hasDefault()) {
```
Comment (oliver-richers) [inline platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/validator/DbTableValidator.java:77]: what is the rationale that nullable fields need a default?

**Review by oliver-richers — CHANGES_REQUESTED** [review]

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/validator/DbTableValidator.java (hunk 2 above)
Comment (alexander-schmidt) [reply platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/validator/DbTableValidator.java:77]: How would I know? Quoting the MySQL docs at https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html#data-type-defaults-implicit:
>  If the column can take NULL as a value, the column is defined with an explicit DEFAULT NULL clause.

Comment (oliver-richers) [reply platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/validator/DbTableValidator.java:72]: pfff, ok ;)

**Review by oliver-richers — APPROVED** [review]

**Pull request merged**